test/choosetests.jl

```diff
@@ -82,60 +82,29 @@ function choosetests(choices = [])
         tests = testnames
     end
 
-
-    unicodetests = ["unicode/utf8"]
-    if "unicode" in skip_tests
-        filter!(x -> (x != "unicode" && !(x in unicodetests)), tests)
-    elseif "unicode" in tests
-        # specifically selected case
-        filter!(x -> x != "unicode", tests)
-        prepend!(tests, unicodetests)
-    end
-
-    stringtests = ["strings/basic", "strings/search", "strings/util",
-                   "strings/io", "strings/types"]
-    if "strings" in skip_tests
-        filter!(x -> (x != "strings" && !(x in stringtests)), tests)
-    elseif "strings" in tests
-        # specifically selected case
-        filter!(x -> x != "strings", tests)
-        prepend!(tests, stringtests)
-    end
-
-    # do subarray before sparse but after linalg
-    if "subarray" in skip_tests
-        filter!(x -> x != "subarray", tests)
-    elseif "subarray" in tests
-        filter!(x -> x != "subarray", tests)
-        prepend!(tests, ["subarray"])
-    end
-
-    compilertests = ["compiler/inference", "compiler/validation", "compiler/ssair", "compiler/irpasses",
-                     "compiler/codegen", "compiler/inline", "compiler/contextual"]
-
-    if "compiler" in skip_tests
-        filter!(x -> (x != "compiler" && !(x in compilertests)), tests)
-    elseif "compiler" in tests
-        # specifically selected case
-        filter!(x -> x != "compiler", tests)
-        prepend!(tests, compilertests)
-    end
-
-    if "stdlib" in skip_tests
-        filter!(x -> (x != "stdlib" && !(x in STDLIBS)) , tests)
-    elseif "stdlib" in tests
-        filter!(x -> (x != "stdlib" && !(x in STDLIBS)) , tests)
-        prepend!(tests, STDLIBS)
-    end
-
-    # do ambiguous first to avoid failing if ambiguities are introduced by other tests
-    if "ambiguous" in skip_tests
-        filter!(x -> x != "ambiguous", tests)
-    elseif "ambiguous" in tests
-        filter!(x -> x != "ambiguous", tests)
-        prepend!(tests, ["ambiguous"])
-    end
-
+    function patchlib!(skip_tests,tests,name,files=[name]; stdlib::Bool=false)
+       flt = x -> (x != name && !(x in files))
+       testsflt = stdlib ? flt : !=(name)
+
+       if name in skip_tests
+           filter!(flt, tests)
+       elseif name in tests
+           filter!(testsflt, tests)
+           prepend!(tests, files)
+       end
+   end
+
+   patchlib!(skip_tests,tests,"unicode", ["unicode/utf8"])
+   patchlib!(skip_tests,tests,"strings", ["strings/basic", "strings/search", "strings/util",
+                  "strings/io", "strings/types"])
+   patchlib!(skip_tests,tests,"subarray")
+   patchlib!(skip_tests,tests,"compiler", ["compiler/inference", "compiler/validation",
+       "compiler/ssair", "compiler/irpasses", "compiler/codegen",
+       "compiler/inline", "compiler/contextual"])
+   patchlib!(skip_tests,tests,"stdlib", STDLIBS; stdlib=true)
+   patchlib!(skip_tests,tests,"ambiguous")
```

Comment:
maybe reproduce here the couple of comments seen before, e.g. `# do ambiguous first to avoid failing if ambiguities are introduced by other tests`

Reply:
Seems to be resolved now